### PR TITLE
chore: Make methods that do not need to be instance methods static

### PR DESCRIPTION
**Pull request by aahung**
…static

#### Why is this change necessary?

There are many instance methods do not need to be instance methods since `self` is not being used. Making it static increases readability. 

#### How does it address the issue?

Making them static. One exception, for `BuildStrategy`, the class is meant to be an abstract class, so I made its two instance methods abstract instead of static. 

#### What side effects does this change have?

no, 
> A static method can be called either on the class (such as C.f()) or on an instance (such as C().f()). https://docs.python.org/3/library/functions.html#staticmethod

We don't need to change how we invoke them.

#### Checklist

- [ ] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [ ] Write design document ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [ ] Write unit tests
- [ ] Write/update functional tests
- [ ] Write/update integration tests
- [x] `make pr` passes
- [ ] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

File: tests/unit/lib/build_module/test_build_strategy.py [hunk 1]
```diff
@@ -39,14 +39,22 @@ def setUp(self):
...
+        return {}
+
+    def build_single_layer_definition(self, layer_definition):
+        return {}
+
+
 class BuildStrategyTest(BuildStrategyBaseTest):
     def test_build_functions_layers(self):
-        build_strategy = BuildStrategy(self.build_graph)
+        build_strategy = _TestBuildStrategy(self.build_graph)
```
Comment (elbayaaa) [inline tests/unit/lib/build_module/test_build_strategy.py:52]: would you explain why we need this  "_TestBuildStrategy" class?

Comment (aahung) [reply tests/unit/lib/build_module/test_build_strategy.py:52]: `BuildStrategy` is an abstract class so I need to create a subclass to instantiate an object.

Comment (aahung) [reply tests/unit/lib/build_module/test_build_strategy.py:52]: (and adding prefix `_` to indicate it is a "private" class)

**Review by elbayaaa — APPROVED** [review]

**Comment by aahung** [comment]
(manually rebase the latest develop to resolve the conflict)

**Comment by aahung** [comment]
Added type hints

**Review by CoshUS — APPROVED** [review]

**Review by elbayaaa — APPROVED** [review]

**Pull request merged**